drivers/mdio/mdio_shell.c

```diff
@@ -98,11 +103,16 @@ static int cmd_mdio_write(const struct shell *sh, size_t argc, char **argv)
 	uint16_t port_addr;
 	int ret;
 
-	ret = parse_device_arg(sh, argv, &dev);
+	ret = parse_device_arg(sh, argc, argv, &dev);
 	if (ret < 0) {
 		return ret;
 	}
 
+	if (argc < 5) {
```

Comment:
I think this and the below new checks are not needed as the commands take no optional arguments and therefore this is already checked by the shell.

https://github.com/zephyrproject-rtos/zephyr/blob/72dc0e5f0af9e3e2406af72e103601672e5bf973/include/zephyr/shell/shell.h#L531-L532